### PR TITLE
[item] frontends

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
https://jira.sng.sk/browse/MG-43

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -72,7 +73,11 @@
 
 Route::group([
     'prefix' => LaravelLocalization::setLocale(),
-    'middleware' => [ 'localeSessionRedirect', 'localizationRedirect' ]
+    'middleware' => [
+        'localeSessionRedirect',
+        'localizationRedirect',
+        ApplyFrontendScope::class,
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply routes/web.php:79]: This middleware doesn't process the header, it only sets a global scope (currently) for items.

File: app/Elasticsearch/Repositories/ItemRepository.php [hunk 2]
```diff
@@ -67,14 +67,21 @@ public function getSuggestions(int $size, string $search, string $locale = null)
...
+                            'multi_match' => [
+                                'query' => $search,
+                                'type' => 'cross_fields',
+                                'fields' => ['identifier', 'title.suggest', 'author.suggest'],
+                                'operator' => 'and',
+                            ],
+                        ],
+                        'filter' => [
+                            ['term' => ['frontend' => config('app.frontend')]],
+                        ],
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app/Elasticsearch/Repositories/ItemRepository.php:81]: Here I've followed how the locale is being switched in laravel. Getting/setting locale is done through application config.

File: docker-compose.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ services:
       REDIS_HOST: redis
 
   mysql:
-    image: mysql:5.7
+    image: mysql:8.0.36
```
Comment (eronisko) [inline docker-compose.yml:29]: Thanks

**Review by eronisko — COMMENTED** [review]
Sorry for the wait with my comments

File: app/Item.php [hunk 4]
```diff
@@ -123,6 +122,7 @@ class Item extends Model implements IndexableModel, TranslatableContract
 
     protected $casts = [
         'colors' => 'json',
+        'frontends' => 'json',
```
Comment (eronisko) [inline app/Item.php:125]: Would this make a meaningful difference?

https://laravel.com/docs/10.x/eloquent-mutators#array-and-json-casting

```suggestion
        'frontends' => 'array',
```

File: database/migrations/2024_02_08_150119_add_frontends_column_to_items_table.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('items', function (Blueprint $table) {
+            $table->json('frontends')->nullable();
+        });
```
[the remark this answers is not in the document]
Comment (eronisko) [reply database/migrations/2024_02_08_150119_add_frontends_column_to_items_table.php:18]: I'm happy either way but with a DB default we could make the column non-nullable (and have one less state/case to worry about)

**Review by eronisko — APPROVED** [review]
I like this much better, thanks

**Pull request merged**